### PR TITLE
Fix Command API registration method in 1.19 for registry-dependent arguments

### PR DESCRIPTION
**Pull request by LambdAurora**
1.19 introduced a breaking change to how registry-dependent arguments are fetched, it requires passing a new class which isn't passed through the command API currently.

This needs more work to fix it on client commands too.

### CODE REVIEW

**Review by LemmaEOF — APPROVED** [review]

**Review by LemmaEOF — APPROVED** [review]

**Comment by LemmaEOF** [comment]
...good job for approving twice gh lmao

**Comment by LambdAurora** [comment]
Client commands have been fixed too now, change includes:
 - breaking the DISPATCHER field into a method as it has to be rebuilt every time the client connects to a server/world,
 - have a similar fix as the server commands.

**Review by Leo40Git — APPROVED** [review]

File: library/command/client_command/src/main/java/org/quiltmc/qsl/command/api/client/ClientCommandManager.java [hunk 1]
```diff
@@ -58,20 +60,23 @@
...
 	/**
-	 * The command dispatcher that handles client command registration and execution.
+	 * {@return the command dispatcher that handles client command registration and execution}
+	 *
+	 * @see ClientCommandRegistrationCallback client command registration
 	 */
-	public static final CommandDispatcher<QuiltClientCommandSource> DISPATCHER = new CommandDispatcher<>();
-
-	private ClientCommandManager() {
+	public static CommandDispatcher<QuiltClientCommandSource> getDispatcher() {
```
Comment (williambl) [inline library/command/client_command/src/main/java/org/quiltmc/qsl/command/api/client/ClientCommandManager.java]: Should have been like this from the start, honestly.

File: library/command/client_command/src/main/java/org/quiltmc/qsl/command/impl/client/ClientCommandInternals.java [hunk 2]
```diff
@@ -90,7 +97,7 @@ public static boolean executeCommand(String message) {
 
 		// The interface is implemented on ClientCommandSource with a mixin.
 		// noinspection ConstantConditions
-		QuiltClientCommandSource commandSource = (QuiltClientCommandSource) client.getNetworkHandler().getCommandSource();
+		var commandSource = (QuiltClientCommandSource) client.getNetworkHandler().getCommandSource();
```
Comment (williambl) [inline library/command/client_command/src/main/java/org/quiltmc/qsl/command/impl/client/ClientCommandInternals.java]: Not pertinent but what's the stance on interface injection in QSL, to eliminate things like this? I saw that it was used once in some other module and then removed after there were issues, are we avoiding it for now?

**Review by williambl — APPROVED** [review]
Looks good.

File: library/command/client_command/src/main/java/org/quiltmc/qsl/command/impl/client/ClientCommandInternals.java (hunk 2 above)
Comment (LambdAurora) [reply library/command/client_command/src/main/java/org/quiltmc/qsl/command/impl/client/ClientCommandInternals.java]: Those issues were fixed, so we can definitely use injected interfaces now

Comment (OroArmor) [reply library/command/client_command/src/main/java/org/quiltmc/qsl/command/impl/client/ClientCommandInternals.java]: Let's get interface injection here and then I'll approve

**Review by OroArmor — COMMENTED** [review]
Add some interface injection

**Review by OroArmor — APPROVED** [review]

**Pull request merged**